### PR TITLE
[SPARK-41010][CONNECT][PYTHON] Complete Support for Except and Intersect in Python client

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

1. Add support for intersect and except.
2. Unify union, intersect and except into `SetOperation`.

### Why are the changes needed?

Improve API coverage.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

UT

### CODE REVIEW

**Comment by amaliujia** [comment]
R: @HyukjinKwon

**Comment by amaliujia** [comment]
actually I will follow https://spark.apache.org/contributing.html. to add a short description for test cases in this PR/

**Comment by amaliujia** [comment]
Ok added short description for the new test cases.

File: python/pyspark/sql/connect/dataframe.py [hunk 1]
```diff
@@ -317,7 +319,83 @@ def unionByName(self, other: "DataFrame", allowMissingColumns: bool = False) ->
...
+        )
+
+    def exceptAll(self, other: "DataFrame") -> "DataFrame":
+        """Return a new :class:`DataFrame` containing rows in this :class:`DataFrame` but
+        not in another :class:`DataFrame` while preserving duplicates.
+
+        This is equivalent to `EXCEPT ALL` in SQL.
+        As standard in SQL, this function resolves columns by position (not by name).
+
+        .. versionadded:: 2.4.0
```
Comment (zhengruifeng) [inline python/pyspark/sql/connect/dataframe.py:335]:
```suggestion
        .. versionadded:: 3.4.0
```

File: python/pyspark/sql/connect/dataframe.py [hunk 2]
```diff
@@ -317,7 +319,83 @@ def unionByName(self, other: "DataFrame", allowMissingColumns: bool = False) ->
...
+        )
+
+    def intersectAll(self, other: "DataFrame") -> "DataFrame":
+        """Return a new :class:`DataFrame` containing rows in both this :class:`DataFrame`
+        and another :class:`DataFrame` while preserving duplicates.
+
+        This is equivalent to `INTERSECT ALL` in SQL. As standard in SQL, this function
+        resolves columns by position (not by name).
+
+        .. versionadded:: 2.4.0
```
Comment (zhengruifeng) [inline python/pyspark/sql/connect/dataframe.py:384]:
```suggestion
        .. versionadded:: 3.4.0
```

**Review by zhengruifeng — APPROVED** [review]

**Comment by zhengruifeng** [comment]
merged into master

File: python/pyspark/sql/connect/plan.py [hunk 3]
```diff
@@ -642,7 +664,7 @@ def _repr_html_(self) -> str:
         return f"""
         <ul>
             <li>
-                <b>Union</b><br />
+                <b>SetOperation</b><br />
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/plan.py:667]: qq: why don't we have Except, Union, etc separately?

Comment (amaliujia) [reply python/pyspark/sql/connect/plan.py:667]: @HyukjinKwon I hope this discussion can answer your question https://github.com/apache/spark/pull/38166#discussion_r991783272....

**Pull request closed without merging**